app/controllers/comments_controller.rb

```diff
@@ -1,17 +1,35 @@
 class CommentsController < ApplicationController
-  load_and_authorize_resource
+  skip_before_action :verify_authenticity_token
+
+  def index
+    @post = Post.find(params[:post_id])
+    @comments = @post.comments.includes(:user)
+    respond_to do |format|
+      format.json { render json: @comments }
+    end
+  end
 
   def new
     @comment = Comment.new
   end
 
   def create
     @post = Post.find(params[:post_id])
-    @comment = @post.comments.new(comment_params)
-    @comment.user = current_user
+    user = current_user || User.last
+
+    @comment = Comment.create(comment_params)
+    @comment.user = user
+    @comment.post = @post
+
     if @comment.save
-      @comment.update_counter
-      redirect_to user_post_path(current_user.id, @post.id), notice: 'You created a comment!'
+      respond_to do |format|
+        format.json do
+          render json: @comment
+        end
+        format.html do
+          redirect_to user_post_path(current_user.id, @post.id), notice: 'You created a comment!'
+        end
+      end
```

Comment:
- Great job 👏🏻 The API endpoint that adds' a comment to a post.  ✔️   

![image](https://github.com/anderCM/blog-app/assets/79058364/855ee338-102f-4e39-91d0-ba8bf486ebac)